components/config-mgmt-service/backend/elastic/runs.go

```diff
@@ -69,6 +71,71 @@ func (es Backend) GetRun(runID string, endTime time.Time) (backend.Run, error) {
 	return run, nil
 }
 
+func (es Backend) GetCheckinCountsTimeSeries(startTime, endTime time.Time,
+	filters map[string][]string) ([]backend.CheckInPeroid, error) {
+	var (
+		dateHistoTag = "dateHisto"
+		mainQuery    = newBoolQueryFromFilters(filters)
+		nodeID       = "node_id"
+	)
+
+	rangeQuery, _ := newRangeQuery(startTime.Format(time.RFC3339),
+		endTime.Format(time.RFC3339), backend.RunEndTime)
+
+	mainQuery = mainQuery.Must(rangeQuery)
+
+	bucketHist := elastic.NewDateHistogramAggregation().Field(backend.RunEndTime).
+		Interval("24h").
+		MinDocCount(0). // needed to return empty buckets
+		ExtendedBounds(
+			startTime.Format(time.RFC3339), endTime.Format(time.RFC3339)). // needed to return empty buckets
+		Format("yyyy-MM-dd'T'HH:mm:ssZ").
+		TimeZone(getTimezoneSoStartAtMidnight(startTime)). // needed start the buckets at the beginning of the day.
```

Comment:
To force elasticsearch to start the buckets at the correct hour the time zone needs to be updated.